hudi-utilities/src/main/java/org/apache/hudi/utilities/streamer/StreamSync.java

```diff
@@ -518,10 +515,10 @@ private Option<String> getLastPendingCompactionInstant(Option<HoodieTimeline> co
   /**
    * Read from Upstream Source and apply transformation if needed.
    *
-   * @return Pair<InputBatch and Boolean> Input data read from upstream source, and boolean is true if empty.
+   * @return Pair<InputBatch and Boolean> Input data read from upstream source, and boolean is true if the result should use the row writer path.
    * @throws Exception in case of any Exception
    */
-  public InputBatch readFromSource(String instantTime, HoodieTableMetaClient metaClient) throws IOException {
+  public Pair<InputBatch, Boolean> readFromSource(String instantTime, HoodieTableMetaClient metaClient) throws IOException {
```

Comment:
this method is also used in hudi-integ-test in `HoodieDeltaStreamerWrapper`. Let's make the change there as well.

Reply:
@codope this is addressed, any other concerns?